### ISSUE TITLE
CLI performance

### ISSUE THREAD

**Issue by SheetJSDev** [infused/dbf#97]
Zipped test file [4MB_35KR.dbf.zip](https://github.com/infused/dbf/files/9124962/4MB_35KR.dbf.zip)

We were surprised to find that the CLI tool that ships with the gem is significantly slower than JS equivalents:

<img width="568" alt="DBF_4MB" src="https://user-images.githubusercontent.com/6070939/179334780-abcde626-abc1-4ba8-b37a-d6ce04db0c57.png">

Is there some more efficient way to generate a CSV from a DBF file?

**Comment by SheetJSDev** [comment]
That helped a bit.  Interestingly, the CSV formatting proved to be a bigger performance hit, with 20% improvement locally by switching from `table.to_csv` to:

```ruby
    table.each do |record|
      puts record.attributes
    end
```

That explains enough to be able to close the issue.

.

One design question: `to_csv` works as follows:

```rb
      each { |record| csv << record.to_a }
```

Reordering the code in `record.rb`: 

```rb
    def attribute_map # :nodoc:
      @columns.map { |column| [column.name, init_attribute(column)] }
    end

    def attributes
      @attributes ||= Hash[attribute_map]
    end

    def to_a
      @columns.map { |column| attributes[column.name] }
    end
```

A hash is constructed from the K/V pairs.  The hash is stored but the K/V pairs are not.  Would it be more efficient/performant in general to store the values array (in column order) and zip the columns and values when the hash is needed?

**Comment by infused** [comment]
Good advice, thank you. This optimization does result in a performance improvement of approximately 10% in the initial testing.

**Comment by infused** [comment]
Released in version 4.2.2
